### ISSUE TITLE
v doc conventions should include using square brackets for v symbols and should generate links

### ISSUE THREAD

**Issue by jeffmikels** [vlang/v#13069]
Consider this code from `vlib/cli/flag.v`

```v
// get_bool returns `true` if the flag specified in `name` is set.
// get_bool returns an error if the `FlagType` is not boolean.
pub fn (flags []Flag) get_bool(name string) ?bool {
	flag := flags.get(name) ?
	return flag.get_bool()
}
```

In the generated documentation `FlagType` should be a link to the specification of the `FlagType` enum.

Because `v doc` already generates HTML anchors for headings, we can do this manually now with markdown as follows:

```v
// get_bool returns `true` if the flag specified in `name` is set.
// get_bool returns an error if the [[FlagType](#FlagType)] is not boolean.
pub fn (flags []Flag) get_bool(name string) ?bool {
	flag := flags.get(name) ?
	return flag.get_bool()
}
```

However, that makes the doc string in the source code more difficult to write and more difficult to read.

In other languages like Dart, they use simple square brackets to tell the documentation generator that something is a keyword. We could do the same and let v doc handle the needed intelligence to format and link properly:

```v
// get_bool returns `true` if the flag specified in `name` is set.
// get_bool returns an error if the [FlagType] is not boolean.
pub fn (flags []Flag) get_bool(name string) ?bool {
	flag := flags.get(name) ?
	return flag.get_bool()
}
```

**Comment by spytheman** [comment]
Good idea.

**Comment by Larpon** [comment]
I like the idea of having syntax for links as described - `[FlagOrFunc]` is a little close to `[attr]` in V code though.

**Comment by JalonSolov** [comment]
Could be confusing if attr's are being documented... which should be a link, and which shouldn't?  Probably need to pick something unused...  `#FlagType#`, maybe?

**Comment by jeffmikels** [comment]
closed in favor of #13099
